app/src/main/java/com/spm/vasylyshyn/counterdisplayapp/activity/MainActivity.kt

```diff
@@ -52,23 +56,22 @@ class MainActivity : AppCompatActivity() {
             if (LoginActivity.token.isNotEmpty()) {
                 val gson = GsonBuilder()
                     .create()
-                val ip = IP.ip
                 val retrofit = Retrofit.Builder()
-                    .baseUrl("http://${ip}:8080/api/")
+                    .baseUrl(URL_PATH)
                     .addConverterFactory(GsonConverterFactory.create(gson))
                     .build()
                 val apiService = retrofit.create(UserService::class.java)
 
-                apiService.getCurrentUser(MainActivity.getHeaderMap())
+                apiService.getCurrentUser(getHeaderMap())
                     .enqueue(object : Callback<User> {
                         override fun onResponse(
                             call: Call<User>,
                             response: Response<User>
                         ) {
                             if (response.isSuccessful) {
-                                username = response.body()!!.username.toString()
-                                phoneNumber = response.body()!!.phoneNumber.toString()
-                                userId = response.body()!!.id.toInt()
+                                username = response.body()?.username.toString()
```

Comment:
![image](https://github.com/vov4ik-v/CounterDisplayApp/assets/74701445/ebd67a82-233b-42e9-b08e-4aaa946074d3)
I see that there are still ```!!``` operators in the project, can you double check them too please? They might be sources of potential app crashes